torch/distributed/checkpoint/filesystem.py

```diff
@@ -349,7 +349,7 @@ def init(self, is_coordinator: bool) -> None:
         pass
 
     def prepare_local_plan(self, plan: SavePlan) -> SavePlan:
-        # There's no storage input in the local plan
+        os.makedirs(self.path, exist_ok=True)
```

Comment:
Looks good, but we should remove from prepare_global_plan as it's now redundant.